xyz-hub-service/src/main/java/com/here/xyz/hub/XYZHubRESTVerticle.java

```diff
@@ -128,13 +131,28 @@
   /**
    * The final response handler.
    */
-  private static void onResponseSent(RoutingContext context) {
+  private static void onResponseEnd(RoutingContext context) {
     final Marker marker = Api.Context.getMarker(context);
+    if (!context.response().headWritten()) {
+      //The response was closed (e.g. by the client) before it could be written
+      logger.info(marker, "The request was cancelled. No response has been sent.");
+      onRequestCancelled(context);
+    }
     logger.info(marker, "{}", LogUtil.responseToLogEntry(context));
     LogUtil.addResponseInfo(context).end();
     LogUtil.writeAccessLog(context);
   }
 
+  private static void onRequestCancelled(RoutingContext context) {
+    context.response().setStatusCode(CLIENT_CLOSED_REQUEST.code());
```

Comment:
I assume these two lines are only necessary for log purposes, as the response is already closed, right?

Reply:
Exactly. The status of the response of VertX is 200 by default right after its instantiation. It would be misleading seeing a "200 OK" for any request which actually has been cancelled.